src/1d/spreadinterp1d.cu

```diff
@@ -0,0 +1,320 @@
+#include <iostream>
+#include <math.h>
+#include <helper_cuda.h>
+#include <cuda.h>
+#include <thrust/extrema.h>
+#include "../../contrib/utils.h"
+#include "../../contrib/utils_fp.h"
+#include "../cuspreadinterp.h"
+#include "../../include/utils.h"
+
+using namespace std;
+
+#define MAXBINSIZE 1024
+
+static __forceinline__ __device__
```

Comment:
These three functions are the same in 1D, 2D, and 3D. Is there a good reason to redefine them here?

Reply:
Not really. I moved them into the header file `cuspreadinterp.h` (see 1d4199e).

Reply:
Great!